### PR TITLE
Fix gcc warning when using map

### PR DESCRIPTION
**Pull request by drivehappy**
Fix for Issue #2211: Addressing GCC warning on enumeral/non-enumeral in conditional expression.

### CODE REVIEW

**Comment by drivehappy** [comment]
I signed it!

**Comment by xfxyjwf** [comment]
ok to test

**Comment by xfxyjwf** [comment]
Can you direct this pull request to 3.1.x branch?

**Comment by haberman** [comment]
@drivehappy Sorry this didn't get merged before. Does this issue still exist? If so, please rebase on master (we're not currently in a release process) and I'll get it merged ASAP!

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

File: src/google/protobuf/map.h [hunk 1]
```diff
@@ -1285,7 +1285,7 @@ class Map {
     // Return a power of two no less than max(kMinTableSize, n).
     // Assumes either n < kMinTableSize or n is a power of two.
     size_type TableSize(size_type n) {
-      return n < kMinTableSize ? kMinTableSize : n;
+      return n < static_cast<size_type>(kMinTableSize) ? static_cast<size_type>(kMinTableSize) : n;
```
Comment (haberman) [inline src/google/protobuf/map.h]: Please wrap at 80 chars per the style guide. Suggested wrapping:

```
    return n < static_cast<size_type>(kMinTableSize)
               ? static_cast<size_type>(kMinTableSize)
               : n;
```

**Comment by drivehappy** [comment]
@haberman: Sorry for the delay, I was out last week - I've made the suggested changes.

Additionally, I'm a bit new to github and don't really know how to rebase to master - I tried it a few months ago, but things seemed to have gotten in a weird/unexpected state, so I moved it back.

**Pull request merged**